src/libponyrt/actor/actor.c

```diff
@@ -15,12 +15,20 @@
 #include <valgrind/helgrind.h>
 #endif
 
+// default actor batch size
+#define PONY_ACTOR_DEFAULT_BATCH 100
+
 // Ignore padding at the end of the type.
 pony_static_assert((offsetof(pony_actor_t, gc) + sizeof(gc_t)) ==
    sizeof(pony_actor_pad_t), "Wrong actor pad size!");
 
 static bool actor_noblock = false;
 
+#define PONY_ACTOR_MUTING_MSGS_NO_SCALING 32 - __pony_clz(PONY_ACTOR_DEFAULT_BATCH)
+
+// default actor throttle factor to ensure muting occurs after 10 msgs sent
+static double actor_throttlefactor = (PONY_ACTOR_MUTING_MSGS_NO_SCALING) / 10.0;
```

Comment:
If I use this patch:
```
diff --git a/src/libponyrt/actor/actor.c b/src/libponyrt/actor/actor.c
index 2c1ea8fb8..22b12c64e 100644
--- a/src/libponyrt/actor/actor.c
+++ b/src/libponyrt/actor/actor.c
@@ -8,6 +8,7 @@
 #include "../gc/trace.h"
 #include "ponyassert.h"
 #include <assert.h>
+#include <stdio.h>
 #include <string.h>
 #include <dtrace.h>
 
@@ -241,6 +242,10 @@ static void try_gc(pony_ctx_t* ctx, pony_actor_t* actor)
 // calculate throttling adjusted batch size
 static size_t throttled_batch_size(pony_actor_t* actor)
 {
+  printf("throttled_batch_size %d\n", (PONY_ACTOR_DEFAULT_BATCH >> (size_t)((atomic_load_explicit(
+    &actor->mute_map_count, memory_order_relaxed) + atomic_load_explicit(
+    &actor->extra_throttled_msgs_sent, memory_order_relaxed)) *
+    actor_throttlefactor)));
   return PONY_ACTOR_DEFAULT_BATCH >> (size_t)((atomic_load_explicit(
     &actor->mute_map_count, memory_order_relaxed) + atomic_load_explicit(
     &actor->extra_throttled_msgs_sent, memory_order_relaxed)) *
@@ -571,6 +569,7 @@ void ponyint_actor_setmsgstilmute(size_t msgs_til_mute)
 
   actor_throttlefactor = (PONY_ACTOR_MUTING_MSGS_NO_SCALING) /
     ((double)msgs_til_mute);
+printf("DBG: PONY_ACTOR_MUTING_MSGS_NO_SCALING %d __pony_clz(PONY_ACTOR_DEFAULT_BATCH) %d actor_throttlefactor %f\n", PONY_ACTOR_MUTING_MSGS_NO_SCALING, __pony_clz(PONY_ACTOR_DEFAULT_BATCH), actor_throttlefactor);
 }
 
 PONY_API pony_actor_t* pony_create(pony_ctx_t* ctx, pony_type_t* type)
```

Then I see that the calculation of `actor_throttlefactor` is wonky, if the default value of the `--ponymsgstilmute` is supposed to be 10; it behaves as if it's 1.  (It's a bit interesting to see how the batch size differs over the short lifetime of each run.)

```
% ./message-ubench --initial-pings 100 --pingers=30 --ponythreads=4 --report-count=1 | & sort | uniq -c
   1 # pingers 30, report-interval 10, report-count 1, initial-pings 100
   1 ,throttled_batch_size 100
   1 1005347000,105075
   1 1550874034.060701000throttled_batch_size 0
   1 DBG: PONY_ACTOR_MUTING_MSGS_NO_SCALING 7 __pony_clz(PONY_ACTOR_DEFAULT_BATCH) 25 actor_throttlefactor 7.000000
15773 throttled_batch_size 0
118613 throttled_batch_size 100
   2 throttled_batch_size 12
   1 time,run-ns,rate

% ./message-ubench --initial-pings 100 --pingers=30 --ponythreads=4 --report-count=1 --ponymsgstilmute=10 | & sort | uniq -c
   1 # pingers 30, report-interval 10, report-count 1, initial-pings 100
   1 ,throttled_batch_size 100
   1 1001977000,118854
   1 1550874071.827285000throttled_batch_size 100
   1 DBG: PONY_ACTOR_MUTING_MSGS_NO_SCALING 7 __pony_clz(PONY_ACTOR_DEFAULT_BATCH) 25 actor_throttlefactor 0.700000
3423 throttled_batch_size 0
5807 throttled_batch_size 1
48089 throttled_batch_size 100
10056 throttled_batch_size 12
23502 throttled_batch_size 25
7098 throttled_batch_size 3
13796 throttled_batch_size 50
17293 throttled_batch_size 6
   1 time,run-ns,rate
```
... and also ...
```
% ./message-ubench --initial-pings 100 --pingers=30 --ponythreads=4 --report-count=1 --ponymsgstilmute=5 | & sort | uniq -c
   1 # pingers 30, report-interval 10, report-count 1, initial-pings 100
   1 ,throttled_batch_size 100
   1 1001182000,115686
   1 1550874104.657073000throttled_batch_size 100
   1 DBG: PONY_ACTOR_MUTING_MSGS_NO_SCALING 7 __pony_clz(PONY_ACTOR_DEFAULT_BATCH) 25 actor_throttlefactor 1.400000
6655 throttled_batch_size 0
55673 throttled_batch_size 100
   1 throttled_batch_size 12
17435 throttled_batch_size 25
13607 throttled_batch_size 3
22780 throttled_batch_size 50
14354 throttled_batch_size 6
   1 time,run-ns,rate

% ./message-ubench --initial-pings 100 --pingers=30 --ponythreads=4 --report-count=1 --ponymsgstilmute=2 | & sort | uniq -c
   1 # pingers 30, report-interval 10, report-count 1, initial-pings 100
   1 ,throttled_batch_size 100
   1 1001254000,111266
   1 1550874125.505395000throttled_batch_size 100
   1 DBG: PONY_ACTOR_MUTING_MSGS_NO_SCALING 7 __pony_clz(PONY_ACTOR_DEFAULT_BATCH) 25 actor_throttlefactor 3.500000
11007 throttled_batch_size 0
   2 throttled_batch_size 1
88502 throttled_batch_size 100
33194 throttled_batch_size 12
   2 throttled_batch_size 25
   1 throttled_batch_size 3
   1 time,run-ns,rate

% ./message-ubench --initial-pings 100 --pingers=30 --ponythreads=4 --report-count=1 --ponymsgstilmute=1 | & sort | uniq -c
   1 # pingers 30, report-interval 10, report-count 1, initial-pings 100
   1 ,throttled_batch_size 100
   1 1001255000,104242
   1 1550874141.137760000throttled_batch_size 100
   1 DBG: PONY_ACTOR_MUTING_MSGS_NO_SCALING 7 __pony_clz(PONY_ACTOR_DEFAULT_BATCH) 25 actor_throttlefactor 7.000000
16069 throttled_batch_size 0
117101 throttled_batch_size 100
   1 time,run-ns,rate
```

Reply:
@slfritchie Ugh. I forgot to set the default for `--ponymsgstilmute` to be `10` so it ends up being `1` and that's why you see that behavior. I'll push a fix to that soon. Thanks for catching the issue.

Reply:
I just pushed a commit to properly set the default for `--ponymsgstilmute`.